src/amp/stores/index.js

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable func-names */
+import { flow, getEnv } from 'mobx-state-tree';
+import base from '../../shared/stores';
+import { home, single } from '../../shared/contexts';
+import { gaVars, gaTriggers } from '../analytics';
+
+export default base.actions(self => ({
+  fetchSelectedItem: flow(function*() {
+    const { connection, settings } = self.root;
+    const selectedItem = getEnv(self).initialSelectedItem;
+
+    if (selectedItem.page) {
+      const { menu } = settings.theme;
+      const context = home(menu);
+      connection.routeChangeSucceed({
+        selectedItem,
+        context,
+      });
+      yield connection.fetchListPage(selectedItem);
+    } else {
+      const context = single();
+      connection.routeChangeSucceed({
+        selectedItem,
+        context,
+      });
+      yield connection.fetchEntity(selectedItem);
```

Comment:
Shouldn't we fetch also the home posts to populate the "Next" button?

Reply:
You are completely right. I'll fix this.

Reply:
Ok, thanks.